spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -324,6 +340,27 @@ def main(argv: Sequence[str]):
         dim = dim_list.index(dim)
         im_out = split_data(im_in, dim)
 
+    elif arguments.stitch:
+        # preserve original orientation (we assume it's consistent among all images)
+        orig_ornt = im_in_list[0].orientation
+
+        # reorient input files and save them to a temp directory
+        path_tmp = tmp_create(basename="image-stitching")
+        fnames_in = []
+        for file in arguments.i:
+            temp_file_path = os.path.join(path_tmp, os.path.basename(file))
+            im_in = Image(file)
+            im_in_rpi = change_orientation(im_in, 'RPI')
+            im_in_rpi.save(temp_file_path, dtype=output_type, verbose=verbose)
+            fnames_in.append(temp_file_path)
+
+        # order fs_names in descending order based on dimensions (largest -> smallest)
+        fnames_in_sorted = sorted(fnames_in, key=lambda fname: max(Image(fname).dim), reverse=True)
```

Comment:
Out of curiosity, what's the rationale for sorting by size? Does it possibly give better stitching results, or maybe faster performance, or maybe it's a good ordering for displaying in the QC report?

Edit: ah, but we use `fnames_in` rather than `fnames_in_sorted` for the QC report, so that's not the reason.

Reply:
The C++ stitching module by Glocket et al. uses the first image as the reference image, i.e. it allocates an array which is filled by the subsequent images along the z-axis. However, it assumes that the images share the x-/y-axis respectively the same dimensions among all images, or that the subsequent images are smaller (and are automatically padded when calling the native C++ "subimage" function). Honestly, I do not know if this design was a deliberate choice, or if our use case never really occurred in their imaging pipeline. However, if we would reverse the order, it would crop a part of the image - so ordering them is crucial :slightly_smiling_face: . I discovered this behavior when testing the feature in the summer.

Reply:
This would be a great piece of information to add to the code comment! Line 357 currently tells us _what_ is happening, but it would be even more useful to know _why_, as you just described. That way, we can know in the future whether the reasons for doing it this way are still valid.

Reply:
I added a small description in #65fc74b :slightly_smiling_face: